### DISCUSSION TITLE
Combine mqtt sensors manually to a device in yaml (Manual device creation)

### DISCUSSION THREAD

**Discussion by Giel538** [home-assistant/architecture#511] [General]
Hi,

I have a lot of custom mqtt sensors which i use. Some of them belong to devices like a dsmr reader, open therm gateway or a device created by my self.  Unfortunately there is no possibility to link this sensors to 1 device, except using mqtt auto discovery (which is not always possible or available). 

I would like to add the possibility to create devices manually and link the sensors to this device.

Something like this should be fine. 

- platform: mqtt
  state_topic: "dsmr/meter-stats/voltage_swell_count_l1"
  name: "dsmr_stats_voltage_swell_count_l1"
  unique_id: "dsmr_stats_voltage_swell_count_l1"
  **device: {"identifiers": ["DSMR"], "name": "DSMR Reader", "model": "DSMR5", "manufacturer": "Test"}**
  unit_of_measurement: ' '
  value_template: "{{ value_json }}"

The line i marked is usually the one you add in auto discovery.

**Comment by emontnemery** [comment]
What's the problem with configuring through MQTT discovery, you can just publish it as a retained message?

**Reply by Giel538** [reply]
> 
> 
> What's the problem with configuring through MQTT discovery, you can just publish it as a retained message?

Yes, off course i can program this for my own scripts. But there are other products which support mqtt but don't have hass.io integration.

**Comment by emontnemery** [comment]
I'm not sure how this is related to hass.io, can you give a concrete example of where it's preferred to setup the devices manually through yaml as opposed to through discovery?
If you setup through yaml, how can you get hold of the device_id?

**Reply by Giel538** [reply]
> I'm not sure how this is related to hass.io, can you give a concrete example of where it's preferred to setup the devices manually through yaml as opposed to through discovery?
> If you setup through yaml, how can you get hold of the device_id?

Well for example the open therm gateway. This unit supports mqtt, but no auto discovery. Most of the time the problem is that the people behind a 3th party platforms have no experience with home assistant and the auto discovery.  So in this case i created all the sensors manually in yaml, which is fine. But i would be nice if i can group them together as one device and attach this device to an area.  And it would also be nice that i can find this device on settings -> integrations -> mqtt  -> manual created devices (for example), to see the corresponding sensors.

If i add this line to all my opentherm sensors:

device: {"identifiers": ["OTGW"], "name": "Opentherm Gateway", "model": "OTGW", "manufacturer": "3th_party"}

the unique id can be created based on the name and the identifier for example. Everytime they match the sensor belongs to the same device, if they dont match a new device has to be created.
